src/block.ts

```diff
@@ -0,0 +1,211 @@
+      bytes: readonly(),
+      value: readonly(),
+      asBlock: readonly()
+    })
+  }
+
+  links (): Iterable<[string, CID<unknown, number, number, API.Version>]> {
+    return links(this.value, [])
+  }
+
+  tree (): Iterable<string> {
+    return tree(this.value, [])
+  }
+
+  get (path = '/'): API.BlockCursorView<unknown> {
+    return get(this.value, path.split('/').filter(Boolean))
+  }
+}
+
+/**
+ * @template T - Logical type of the data encoded in the block
+ * @template Code - multicodec code corresponding to codec used to encode the block
+ * @template Alg - multicodec code corresponding to the hashing algorithm used in CID creation.
+ */
+async function encode <T, Code extends number, Alg extends number> ({ value, codec, hasher }: { value: T, codec: API.BlockEncoder<Code, T>, hasher: API.MultihashHasher<Alg> }): Promise<API.BlockView<T, Code, Alg>> {
+  if (typeof value === 'undefined') throw new Error('Missing required argument "value"')
+  if (codec == null || hasher == null) throw new Error('Missing required argument: codec or hasher')
+
+  const bytes = codec.encode(value)
+  const hash = await hasher.digest(bytes)
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+  const cid = CID.create(
+    1,
+    codec.code,
+    hash
+  ) as CID<T, Code, Alg, 1>
+
+  return new Block({ value, bytes, cid })
+}
+
+/**
+ * @template T - Logical type of the data encoded in the block
+ * @template Code - multicodec code corresponding to codec used to encode the block
+ * @template Alg - multicodec code corresponding to the hashing algorithm used in CID creation.
+ */
+async function decode <T, Code extends number, Alg extends number> ({ bytes, codec, hasher }: { bytes: API.ByteView<T>, codec: API.BlockDecoder<Code, T>, hasher: API.MultihashHasher<Alg> }): Promise<API.BlockView<T, Code, Alg>> {
+  if (bytes == null) throw new Error('Missing required argument "bytes"')
+  if (codec == null || hasher == null) throw new Error('Missing required argument: codec or hasher')
+
+  const value = codec.decode(bytes)
+  const hash = await hasher.digest(bytes)
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+  const cid = CID.create(1, codec.code, hash) as CID<T, Code, Alg, 1>
+
+  return new Block({ value, bytes, cid })
+}
+
+/**
+ * @template T - Logical type of the data encoded in the block
+ * @template Code - multicodec code corresponding to codec used to encode the block
+ * @template Alg - multicodec code corresponding to the hashing algorithm used in CID creation.
+ * @template V - CID version
+ */
+function createUnsafe <T, Code extends number, Alg extends number, V extends API.Version> ({ bytes, cid, value: maybeValue, codec }: { cid: API.Link<T, Code, Alg, V>, value: T, codec?: API.BlockDecoder<Code, T>, bytes: API.ByteView<T> } | { cid: API.Link<T, Code, Alg, V>, bytes: API.ByteView<T>, value?: undefined, codec: API.BlockDecoder<Code, T> }): API.BlockView<T, Code, Alg, V> {
```

Comment:
wild, the function signatures in here are huge! could we extract some types from here to simplify? For reference, we've gone from the admittedly verbose:

```js
 * @param {{ cid: API.Link<T, Code, Alg, V>, value:T, codec?: API.BlockDecoder<Code, T>, bytes: API.ByteView<T> }|{cid:API.Link<T, Code, Alg, V>, bytes:API.ByteView<T>, value?:void, codec:API.BlockDecoder<Code, T>}} options
 * @returns {API.BlockView<T, Code, Alg, V>}
 */
function createUnsafe ({ bytes, cid, value: maybeValue, codec }) {
```

to shoving it all on a single line.

A rule of thumb I try to follow is to break at 80 characters unless that's too awkward (I have lines in my editor at 80 and 100 chars to help me manually stick to this, roughly). In a case like this, linebreaks would certainly help make this more readable, but maybe attempting to go back a bit to the multipart definition like we had before with this spread out over 3 separate places, would be good. Can we define a private (? local? something that doesn't get in the way of the public consumption of this) form that makes this more compact, name a type or 3 and combine them in a more compact form?

Reply:
I'll split these long lines up better using local types. Maybe we could also update linter rules upstream to warn or error on >100 chars? Not sure what the current rule is.

Reply:
@wemeetagain are you still going to make the requested changes?

Reply:
I've made the changes